misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ActionNodeVisitor.java

```diff
@@ -86,7 +86,18 @@ public List<Interaction> getInteractionList() {
 
     @Override
     public void visit(ClientResourceAccessActionNode clientResourceAccessActionNode) {
-        ExpressionNode clientNode = clientResourceAccessActionNode.expression();
+        SimpleNameReferenceNode clientNode = null;
+        if (clientResourceAccessActionNode.expression() instanceof FieldAccessExpressionNode) {
+            NameReferenceNode fieldName = ((FieldAccessExpressionNode)
+                    clientResourceAccessActionNode.expression()).fieldName();
+            if (fieldName instanceof SimpleNameReferenceNode) {
+                clientNode = (SimpleNameReferenceNode) fieldName;
+            }
+            // todo : Other combinations
+        } else if (clientResourceAccessActionNode.expression() instanceof SimpleNameReferenceNode) {
```

Comment:
@AzeemMuzammil shouldn't we ideally make this field access expression logic as recursive, to handle field access expressions with chain access?